### PR TITLE
feat: Virtualised list for search results

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Wraps the search results in a [virtualised list](https://www.patterns.dev/vanilla/virtual-lists/), using [react-virtuoso](https://virtuoso.dev/)
 - Restructures hierarchy of components in order to fit the pattern required by Virtuoso - 
   - Removes `SearchNodeResult` component - this is now just an item within the list ([code](https://github.com/theopensystemslab/planx-new/pull/3722/files#diff-f3cf32f5f5d3b76d1131248ad552fbf5f8e7fc1d4dde644da0ca81d449d62cceR110))
   - Split inputs into `SearchHeader`
   - Retain logic and component state in `Search`
 - Updates tests to match the new pattern (wrapping in Virtuosos and setting up context)

## What's changed?
Previously, performance felt sluggish when a large number of search results were returned. Scrolling was slow, and the text input would frequently freeze. This is mitigated by using a debounce for the search term, but still noticeable.

Please see this example flow, with search term "user" (215 results) - https://editor.planx.dev/opensystemslab/prior-approval-scope-check

On the Pizza, this should feel a fair bit quicker - we can see in the video below that only a small number of result cards are being rendered at any given moment.

https://github.com/user-attachments/assets/d0c3cd19-26dd-4ac6-b818-eefcc1938ca8

### CODE REVIEW

File: editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/index.tsx [hunk 1]
```diff
@@ -50,72 +85,30 @@ const Search: React.FC = () => {
...
+        isSearching,
+        lastPattern,
+        formik,
+        setIsSearching,
+      }}
+      components={{
+        Footer: ExternalPortalList,
+        List: ListComponent,
+        Item: ListItemComponent,
+        Header: SearchHeader,
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/index.tsx:108]: Child components are passed in via the `components` prop ([docs](https://virtuoso.dev/footer/)).

File: editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/index.tsx [hunk 2]
```diff
@@ -21,25 +24,57 @@ interface SearchNodes {
...
+const ListComponent = React.forwardRef<HTMLUListElement>((props, ref) => (
+  <List {...props} ref={ref} sx={{ mx: 3 }} />
+)) as Components<Data, Context>["List"];
+
+/**
+ * Accessibility - Render the Virtuoso item as a HTMLLiElement, not a HTMLDivElement
+ */
+const ListItemComponent = React.forwardRef<HTMLLIElement>((props, ref) => (
+  <ListItem disablePadding sx={{ pb: 2 }} {...props} ref={ref} />
+)) as Components<Data, Context>["Item"];
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/components/Sidebar/Search/index.tsx:49]: I couldn't find a way to set this up without casting to Virtuoso types.

File: editor.planx.uk/package.json [hunk 3]
```diff
@@ -81,6 +81,7 @@
     "react-navi-helmet-async": "^0.15.0",
     "react-toastify": "^9.1.3",
     "react-use": "^17.5.0",
+    "react-virtuoso": "^4.10.4",
```
Comment (DafyddLlyr) [inline editor.planx.uk/package.json:84]: **Why Virtuoso?**
 - Recommended in [MUI docs](https://mui.com/material-ui/react-list/#virtualized-list)
 - I tried the more basic react-window first and quickly found myself having a lot of CSS and layout issues - Virtuoso "just worked"
 - Good score on [socket.dev](https://socket.dev/npm/package/react-virtuoso)
 - [Docs](https://virtuoso.dev/) were solid and clear

**Review by jessicamcinchak — APPROVED** [review]
Feels like a solid improvement ! Appreciate the why-react-virtuoso decision making notes

Only noticed a few things _not_ related to the virtulised table changes while reviewing which I've dropped in the Slack feedback thread, this seems good to go !

**Comment by DafyddLlyr** [comment]
Thanks @jessicamcinchak - working my way through those comments now 👍

**Pull request merged**